### PR TITLE
[RAPTOR-5929] Functional test coverage for schema validation failing

### PR DESCRIPTION
**Pull request by jmbannon**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Adds two functional tests: input and output schema validation failing

### CODE REVIEW

File: tests/functional/test_custom_task_templates.py [hunk 1]
```diff
@@ -206,3 +219,83 @@ def test_custom_task_templates(
...
+            ),
+            (
+                "fixture",
+                "validate_transform_fail_output_schema_validation",
+                "project_binary_diabetes",
+                "sklearn_drop_in_env",
+                "transform",
+                [
+                    "Schema validation found mismatch between dataset and the supplied schema",
+                    "Datatypes incorrect. Data has types: NUM, but expected types to exactly match: CAT",
```
Comment (jmbannon) [inline tests/functional/test_custom_task_templates.py:245]: I'm going to update this to make it more obvious it's the output data

File: tests/functional/test_custom_task_templates.py [hunk 2]
```diff
@@ -206,3 +219,83 @@ def test_custom_task_templates(
...
+        )
+
+        # Only use numeric input features
+        w = Workshop()
+        bp = w.CustomTask(custom_task_version.custom_task_id, version=str(custom_task_version.id))(
+            w.TaskInputs.NUM
+        )
+        bp = w.Tasks.LR1()(bp)
+
+        user_blueprint = w.BlueprintGraph(bp).save()
```
Comment (jmbannon) [inline tests/functional/test_custom_task_templates.py:273]: Also going to add an assert on blueprint validation - we should be able to see the input/output types at this stage

Comment (jmbannon) [reply tests/functional/test_custom_task_templates.py:273]: BP workshop only supports jupyter notebook validation atm

**Review by scottp-ml — APPROVED** [review]
LGTM

**Review by meltzg — APPROVED** [review]

**Pull request merged**